CMakeLists.txt

```diff
@@ -1,6 +1,105 @@
-# CMakeLists.txt has to be located in the project folder and cmake has to be
-# executed from 'project/build' with 'cmake ../'.
-cmake_minimum_required(VERSION 2.6)
-find_package(Rock)
-rock_init(vizkit3d_normal_depth_map 0.1)
-rock_standard_layout()
+cmake_minimum_required(VERSION 2.8.3)
+project(gazebo_ros_sonar_plugin)
+
+
+find_package(catkin REQUIRED COMPONENTS roslint
+  image_transport 
+  roscpp 
+  sensor_msgs)
+find_package(OpenCV REQUIRED)
+find_package(GAZEBO REQUIRED)
+find_package(Boost REQUIRED)
+
+set(GAZEBO_ROS_SONAR_PLUGIN "")
+
+catkin_package(
+  INCLUDE_DIRS include
+    ${GAZEBO_INCLUDE_DIRS}
+    ${GAZEBO_MSG_INCLUDE_DIRS}
+    ${Boost_INCLUDE_DIRS}
+    ${catkin_INCLUDE_DIRS}
+)
+
+include_directories(
+  ${PROJECT_SOURCE_DIR}/include
+  ${PROJECT_SOURCE_DIR}/test/include
+  ${catkin_INCLUDE_DIRS}
+  ${GAZEBO_INCLUDE_DIRS}
+  ${GAZEBO_ODE_INCLUDE_DIRS}
+  ${GAZEBO_MSG_INCLUDE_DIRS}
+  ${Boost_INCLUDE_DIRS}
+  ${OpenCV_INCLUDE_DIRS}
+)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GAZEBO_CXX_FLAGS}")
+set(ROSLINT_CPP_OPTS "--extensions=hpp,cpp,c,hh,cc,h")
+
+set(GAZEBO_ROS_SONAR_PLUGIN_SRCS
+  src/FLSonarRos.cc)
+set(GAZEBO_SONAR_PLUGIN_SRCS
+  src/FLSonar.cc)
+
+set(GAZEBO_ROS_SONAR_PLUGIN_HEADERS
+ include/${PROJECT_NAME}/FLSonar.hh)
+
+roslint_cpp()
+
+roslint_cpp(${GAZEBO_ROS_SONAR_PLUGIN_SRCS}
+  ${GAZEBO_ROS_SONAR_PLUGIN_HEADERS})
+
+add_library(FLSonar ${GAZEBO_SONAR_PLUGIN_SRCS})
+target_link_libraries(FLSonar ${GAZEBO_LIBRARIES} ${OpenCV_LIBRARIES})
+list(APPEND GAZEBO_SONAR_PLUGIN_LIST FLSonar)
+
+add_library(fl_sonar_ros ${GAZEBO_ROS_SONAR_PLUGIN_SRCS})
+target_link_libraries(fl_sonar_ros ${catkin_LIBRARIES} ${GAZEBO_SONAR_PLUGIN_LIST})
+add_dependencies(fl_sonar_ros ${catkin_EXPORTED_TARGETS})
+list(APPEND GAZEBO_ROS_SONAR_PLUGIN_LIST fl_sonar_ros)
```

Comment:
Why does one plugin use PascalCase naming and the other one use snake_case naming? Gazebo base plugins use PascalCase, you should probably stick to that..